rclpy/src/rclpy/_rclpy.c

```diff
@@ -2046,6 +2041,20 @@ rclpy_destroy_node_entity(PyObject * Py_UNUSED(self), PyObject * args)
   Py_RETURN_NONE;
 }
 
+/// Destructor for a clock
+void
+_rclpy_destroy_clock(PyObject * pycapsule)
+{
+  rcl_clock_t * clock = (rcl_clock_t *)PyCapsule_GetPointer(pycapsule, "rcl_clock_t");
```

Comment:
The code should check for `!clock`.

Reply:
https://github.com/ros2/rclpy/pull/212/commits/d82ca21d7c11e41d42b553791b2365698fed3715

Reply:
So currently it silently returns?

Reply:
Sorry, I still don't understand it. If `clock` is `NULL` that means a wrong parameter was passed - not a `rcl_clock_t` capsule. How is this early return different from the case where `clock` is not `null` and correctly finalized?

Reply:
> How is this early return different from the case where clock is not null and correctly finalized?

If `PyCapsule_GetPointer()` returns `NULL` then it raised an exception. If the clock is correctly finalized then no exception is raised.

>  If clock is NULL that means a wrong parameter was passed

I don't see a way to pass a wrong parameter to this function at the moment. It shouldn't get the wrong type of pointer when using it as a capsule destructor.

https://github.com/ros2/rclpy/blob/5ed7ca2755ad8dbee293858f50f2a2707342624d/rclpy/src/rclpy/_rclpy.c#L3184

And the new call in this PR calls `PyCapsule_IsValid()` first. `PyCapsule_GetPointer()` is guaranteed not to return `NULL` if that passes.

Reply:
Thanks for the detailed description. I guess I was kind of confused by the comment - I updated it in a new commit: a6934ca5013637942f75c23150af8597c2f99343